instrumentation/spring/spring-web/spring-web-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springweb/v3_1/SpringWebInstrumentationModule.java

```diff
@@ -31,13 +32,9 @@ public ElementMatcher.Junction<ClassLoader> classLoaderMatcher() {
   }
 
   @Override
-  public void registerHelperResources(HelperResourceBuilder helperResourceBuilder) {
```

Comment:
How do you know that this is safe to remove? As far as I can tell this code was added without a test.

Reply:
You are right, I forgot to check that no test was added with https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7447 PR, however the proxy class is injected both as a class and resource so I assumed that it was enough here.

There was a way to reproduce in https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7428 so maybe it could be used to at least validate it manually once, do you think that could be enough here ? Covering that with a proper test might be tedious, so if you have ideas it I'm definitely open to suggestions :-)

Reply:
I think relying on the proxy class adding a resource is fine. Replicating the original issue seems like too much effort.

Reply:
We still need to keep this for the non-indy case. Could surround it with `if (!isIndyModule())`

Reply:
Fixed in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12432/commits/a5806c600bb35e202a191543631d32bf334b421f